diff --git a/org/eclipse/osgi/internal/serviceregistry/ServiceRegistry.java b/org/eclipse/osgi/internal/serviceregistry/ServiceRegistry.java
index fe014c2d1c7cb..c6cccab5544f0 100644
--- a/org/eclipse/osgi/internal/serviceregistry/ServiceRegistry.java
+++ b/org/eclipse/osgi/internal/serviceregistry/ServiceRegistry.java
@@ -886,30 +886,35 @@ public class ServiceRegistry {
 			for (Map.Entry<BundleContextImpl, CopyOnWriteIdentityMap<ServiceListener, FilteredServiceListener>> bundleContextEntry :
 					listenerSnapshot.entrySet()) {
 
+				BundleContextImpl bundleContextImpl = bundleContextEntry.getKey();
+
+				Bundle bundle = bundleContextImpl.getBundleImpl();
+
 				CopyOnWriteIdentityMap<ServiceListener, FilteredServiceListener> map = bundleContextEntry.getValue();
 
 				for (FilteredServiceListener filteredServiceListener : map.values()) {
 					try {
 						filteredServiceListener.serviceChanged(event);
 					}
-					catch (Throwable t) {
-						if (debug.DEBUG_GENERAL) {
-							Debug.println(
-								"Exception in bottom level event dispatcher: " +
-									t.getMessage());
+					catch (Throwable t1) {
+						Debug.println("Exception in bottom level event dispatcher: " + t1.getMessage());
 
-							Debug.printStackTrace(t);
-						}
+						Debug.printStackTrace(t1);
 
-						container.handleRuntimeError(t);
+						container.handleRuntimeError(t1);
 
-						EquinoxEventPublisher equinoxEventPublisher =
-							container.getEventPublisher();
+						try {
+							EquinoxEventPublisher equinoxEventPublisher = container.getEventPublisher();
 
-						BundleContextImpl bundleContextImpl = bundleContextEntry.getKey();
+							equinoxEventPublisher.publishFrameworkEvent(FrameworkEvent.ERROR, bundle, t1);
+						}
+						catch (Throwable t2) {
+							Debug.println("Exception in bottom level event dispatcher: " + t2.getMessage());
+
+							Debug.printStackTrace(t2);
 
-						equinoxEventPublisher.publishFrameworkEvent(
-							FrameworkEvent.ERROR, bundleContextImpl.getBundle(), t);
+							container.handleRuntimeError(t2);
+						}
 					}
 				}
 			}
