diff --git a/org/springframework/web/util/UriComponentsBuilder.java b/org/springframework/web/util/UriComponentsBuilder.java
index 0c79c2fa95cee..216b823489b6e 100644
--- a/org/springframework/web/util/UriComponentsBuilder.java
+++ b/org/springframework/web/util/UriComponentsBuilder.java
@@ -74,7 +74,7 @@ public class UriComponentsBuilder implements UriBuilder, Cloneable {
 
 	private static final String HTTP_PATTERN = "(?i)(http|https):";
 
-	private static final String USERINFO_PATTERN = "([^@\\[/?#]*)";
+	private static final String USERINFO_PATTERN = "([^@/?#]*)";
 
 	private static final String HOST_IPV4_PATTERN = "[^\\[/?#:]*";
 
@@ -1004,3 +1004,4 @@ public class UriComponentsBuilder implements UriBuilder, Cloneable {
 	private enum EncodingHint { ENCODE_TEMPLATE, FULLY_ENCODED, NONE }
 
 }
+/* @generated */
\ No newline at end of file
