diff --git a/org/apache/axis/client/ServiceFactory.java b/org/apache/axis/client/ServiceFactory.java
index bd0a41de8cbc..a42c58eee7db 100644
--- a/org/apache/axis/client/ServiceFactory.java
+++ b/org/apache/axis/client/ServiceFactory.java
@@ -106,6 +106,17 @@ public class ServiceFactory extends javax.xml.rpc.ServiceFactory
         
         if (context != null) {
             String name = (String)environment.get("jndiName");
+
+            if(name!=null && (name.toUpperCase().indexOf("LDAP")!=-1 ||
+                    name.toUpperCase().indexOf("RMI")!=-1 ||
+                    name.toUpperCase().indexOf("JMS")!=-1 ||
+                    name.toUpperCase().indexOf("JMX")!=-1) ||
+                    name.toUpperCase().indexOf("JRMP")!=-1 ||
+                    name.toUpperCase().indexOf("JAVA")!=-1 ||
+                    name.toUpperCase().indexOf("DNS")!=-1)  {
+
+                return null;
+            }
             if (name == null) {
                 name = "axisServiceName";
             }
@@ -120,6 +131,7 @@ public class ServiceFactory extends javax.xml.rpc.ServiceFactory
                     context.bind(name, service);
                 } catch (NamingException e1) {
                     // !!! Couldn't do it, what should we do here?
+                    return null;
                 }
             }
         } else {
